modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapper.java

```diff
@@ -107,23 +123,111 @@ public boolean isAssignableFrom(Class<?> clazz) {
     return aClass != null && aClass.isAssignableFrom(clazz);
   }
 
+  @Override
+  public boolean isAssignableFrom(Type type) {
+    if (type instanceof TypeWrapper) {
+      return type.getDeclaringClass().get().isAssignableFrom(aClass);
+    } else if (type instanceof ASTType) {
+      return type.isAssignableTo(this);
+    }
+    return false;
+  }
+
   @Override
   public boolean isAssignableTo(Class<?> clazz) {
     return aClass != null && clazz.isAssignableFrom(aClass);
   }
 
   @Override
-  public java.lang.reflect.Type getReflectType() {
-    return type;
+  public boolean isAssignableTo(Type type) {
+    return type.isAssignableFrom(this);
+  }
+
+  @Override
+  public boolean isSameType(Type type) {
+    if (type instanceof TypeWrapper) {
+      Class<?> aClass = ((TypeWrapper) type).aClass;
+      return type.isSameType(aClass);
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isSameType(Class<?> clazz) {
+    return aClass.equals(clazz);
+  }
+
+  @Override
+  public boolean isInstantiable() {
+    return IntrospectionUtils.isInstantiable(aClass);
   }
 
   @Override
   public String getTypeName() {
     return aClass.getTypeName();
   }
 
+  @Override
+  public ClassInformationAnnotation getClassInformation() {
+    return new ClassInformationAnnotation(aClass, of(resolvableTypeGenerics)
+        .map(ResolvableType::getType)
+        .collect(toList()));
+  }
+
+  @Override
+  public boolean isAnyType() {
```

Comment:
isn't this information already in the metadataType?

Reply:
same as explained above, this is a hack and the information can only be obtained from the java type